wasm/src/main/scala/org/scalajs/linker/backend/wasmemitter/CoreWasmLib.scala

```diff
@@ -2166,4 +2200,106 @@ object CoreWasmLib {
     fb.buildAndAddToModule()
   }
 
+  private def genAllocate()(implicit ctx: WasmContext): Unit = {
```

Comment:
What allocate does is just shifting the `currentAddress` based on the given `size` of bytes. If the current memory size (by `memory.size`) is in short, we call `memory.grow` (if it returns `-1` it means reached to the max, throwing an exception).